dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/PoolingLayer.java

```diff
@@ -76,19 +81,75 @@ public boolean isLearnable() {
     return false;
   }
 
+  /**
+   * Feedforward function for max pooling.
+   * @param input the input values for this layer.
+   * @return the output values for this layer.
+   */
   private Matrix feedForwardMaxPooling(final Matrix input) {
-    throw new RuntimeException("Not implemented");
+    final int[] inputShape = getInputShape();
+    final Matrix output = matrixFactory.create(outputShape[0] * outputShape[1], input.getColumns());
+    indexMatrix = matrixFactory.create(outputShape[0] * outputShape[1], input.getColumns());
```

Comment:
The product of output shape dimensions can be calculated once and we can use it repeatedly.
In addition, we don't have to create a new `indexMatrix` every time. Why don't we generate `indexMatrix` in the constructor and just update its elements? If we do, `indexMatrix` should be `final`. This can remove unnecessary memory allocation.